src/cycles.ts

```diff
@@ -0,0 +1,92 @@
+import {LineSegment, Point, SpaceEfficientAdjacencyMatrix} from './line-tools';
+
+/**
+ * Interface for connected cycles.
+ * @interface
+ * @property {LineSegment[][]} cycles - An array of cycles, where each cycle is an array of line segments.
+ */
+interface ConnectedCycle {
+    cycles: LineSegment[][];
+}
+
+/**
+ * Finds all cycles in a given graph.
+ * @function findCycles
+ * @param {LineSegment[]} graph - An array of line segments representing the graph.
+ * @returns {ConnectedCycle[]} - An array of connected cycles found in the graph.
+ */
+export function findCycles(graph: LineSegment[]): ConnectedCycle[] {
+    const matrix = new SpaceEfficientAdjacencyMatrix(graph);
```

Comment:
Don't use the matrix here, just create UniquePoints. In the _dfsForCycles, you are iterating through each point in the matrix, even ones that don't have a connection, which makes the spaceefficientmatrix basically useless. Just use a regular matrix.